model/map-jpa/src/main/resources/META-INF/users/jpa-users-changelog-1.xml

```diff
@@ -0,0 +1,180 @@
+            <column name="fk_root"/>
+        </createIndex>
+        <createIndex tableName="kc_user_consent" indexName="user_consent_clientId">
+            <column name="clientid"/>
+        </createIndex>
+        <ext:createJsonIndex tableName="kc_user_consent" indexName="user_consent_grantedClientScopes">
+            <ext:column jsonColumn="metadata" jsonProperty="fGrantedClientScopesIds"/>
+        </ext:createJsonIndex>
+
+        <createTable tableName="kc_user_federated_identity">
+            <column name="id" type="UUID">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="version" type="INTEGER" defaultValueNumeric="0">
+                <constraints nullable="false"/>
+            </column>
+            <column name="fk_root" type="UUID">
+                <constraints foreignKeyName="user_federated_identity_fk_root_fkey" references="kc_user(id)" deleteCascade="true"/>
+            </column>
+            <column name="metadata" type="json"/>
+        </createTable>
+        <ext:addGeneratedColumn tableName="kc_user_federated_identity">
+            <ext:column name="identityprovider" type="VARCHAR(255)" jsonColumn="metadata" jsonProperty="fIdentityProvider"/>
+            <ext:column name="userid" type="KC_KEY" jsonColumn="metadata" jsonProperty="fUserId"/>
+        </ext:addGeneratedColumn>
+        <createIndex tableName="kc_user_federated_identity" indexName="user_federated_identity_fk_root">
+            <column name="fk_root"/>
+        </createIndex>
+        <createIndex tableName="kc_user_federated_identity" indexName="user_federated_identity_idp">
+            <column name="identityprovider"/>
+        </createIndex>
+        <createIndex tableName="kc_user_federated_identity" indexName="user_federated_identity_userId">
+            <column name="userid"/>
+        </createIndex>
+
+        <createTable tableName="kc_user_group">
+            <column name="user_id" type="UUID">
+                <constraints nullable="false" foreignKeyName="user_group_userid_fkey" references="kc_user(id)" deleteCascade="true"/>
+            </column>
+            <column name="group_id" type="KC_KEY"/>
+        </createTable>
+        <createIndex tableName="kc_user_group" indexName="user_group_userId">
+            <column name="user_id"/>
+        </createIndex>
+        <createIndex tableName="kc_user_group" indexName="user_group_groupId">
+            <column name="group_id"/>
+        </createIndex>
+
+        <createTable tableName="kc_user_role">
+            <column name="user_id" type="UUID">
+                <constraints nullable="false" foreignKeyName="user_role_userid_fkey" references="kc_user(id)" deleteCascade="true"/>
+            </column>
+            <column name="role_id" type="KC_KEY"/>
+        </createTable>
+        <createIndex tableName="kc_user_role" indexName="user_role_userId">
+            <column name="user_id"/>
```

Comment:
Looking at this again today, I found `user_id` and I thought if that should be `userid` without the underscore. I also noticed that I used `role_id` in my other PR. Do you have an opinion on this? I'm approving it anyway, as if this would need to change, it would also need to change in the other PR as well.

Reply:
Funny you mentioned this because yesterday I was looking into some of our schemas and I noticed the very same inconsistency in naming the columns. I personally like it better with underscores, but I don't have a strong opinion on which format should be used. On the other hand I really think we should review the schemas and go for a single and unified format for the column names.

Reply:
I've created https://github.com/keycloak/keycloak/issues/12929 to follow up on this.